### PR TITLE
Add collection:searchSpecifications

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR adds `collection:searchSpecifications` with its units tests.

Sample code :

```java
import io.kuzzle.sdk.*;
import io.kuzzle.sdk.CoreClasses.Responses.Response;
import io.kuzzle.sdk.CoreClasses.SearchResult;
import io.kuzzle.sdk.Options.SearchOptions;
import io.kuzzle.sdk.Options.KuzzleOptions;
import io.kuzzle.sdk.Options.Protocol.WebSocketOptions;
import io.kuzzle.sdk.Options.SearchOptions;
import io.kuzzle.sdk.Protocol.WebSocket;

import java.util.ArrayList;
import java.util.concurrent.ConcurrentHashMap;

public class searchSpecifications {
    private static Kuzzle kuzzle;

    public static void main(String[] args) {
        WebSocketOptions opts = new WebSocketOptions();
        opts.setAutoReconnect(true).setConnectionTimeout(42000);

        try {
            WebSocket ws = new WebSocket("localhost", opts);

            kuzzle = new Kuzzle(ws, (KuzzleOptions) null);
            kuzzle.connect();
            

            ConcurrentHashMap<String, Object> searchQuery = new ConcurrentHashMap<>();
            ConcurrentHashMap<String, Object> filters = new ConcurrentHashMap<>();
            ConcurrentHashMap<String, Object> boost = new ConcurrentHashMap<>();

            boost.put("boost", 1);
            filters.put("match_all", boost);
            searchQuery.put("query", filters);

            SearchOptions options = new SearchOptions();
            options.setSize(50);
            options.setFrom(0);
            options.setScroll("10s");

            SearchResult result = kuzzle
                 .getCollectionController()
                 .searchSpecifications(searchQuery, options).get();

            System.out.println(result.hits);
        }  catch (Exception e) {
            e.printStackTrace();
        }
        kuzzle.disconnect();
    }
};
```

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### Other changes
Correct how `refresh` option is passed and adapt tests

### CODE REVIEW

File: doc/3/controllers/collection/search-specifications/index.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 code: true
 type: page
+<<<<<<< HEAD
```
Comment (Aschen) [inline doc/3/controllers/collection/search-specifications/index.md]: Merge leftover

File: doc/3/core-classes/search-result/next/index.md [hunk 2]
```diff
@@ -50,7 +50,11 @@ It can lead to memory leaks if ascroll duration too great is provided, or if too
 You can restrict the scroll session maximum duration under the `services.storage.maxScrollDuration` configuration key.
 :::
 
+<<<<<<< HEAD
```
Comment (Aschen) [inline doc/3/core-classes/search-result/next/index.md]: Merge leftover

File: doc/3/controllers/collection/search-specifications/index.md (hunk 1 above)
Comment (Yoann-Abbes) [reply doc/3/controllers/collection/search-specifications/index.md]: I think you are not up to date, check the last commit

File: doc/3/core-classes/search-result/next/index.md (hunk 2 above)
Comment (Yoann-Abbes) [reply doc/3/core-classes/search-result/next/index.md]: same as above

File: doc/3/controllers/collection/search-specifications/snippets/search-specifications.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+
+ConcurrentHashMap<String, Object> searchQuery = new ConcurrentHashMap<>();
+ConcurrentHashMap<String, Object> filters = new ConcurrentHashMap<>();
+ConcurrentHashMap<String, Object> boost = new ConcurrentHashMap<>();
+
+boost.put("boost", 1);
```
Comment (Aschen) [inline doc/3/controllers/collection/search-specifications/snippets/search-specifications.java:6]: Why do you use `boost` ?

Comment (Aschen) [reply doc/3/controllers/collection/search-specifications/snippets/search-specifications.java:6]: IMHO you can remove it, it will confuse the user I think

**Pull request merged**